clash-cores/src/Clash/Cores/Sgmii/Sync.hs

```diff
@@ -0,0 +1,284 @@
+{-# LANGUAGE RecordWildCards #-}
+
+module Clash.Cores.Sgmii.Sync where
+
+import Clash.Cores.Sgmii.Common
+import Clash.Cores.Sgmii.EbTb
+import Clash.Prelude
+
+-- | State type of 'sync'. This contains all states as they are defined in IEEE
+--   802.3 Clause 36.
+data SyncState
+  = LossOfSync {_rd :: Bool, _rxEven :: RxEven}
+  | CommaDetect1 {_rd :: Bool, _rxEven :: RxEven}
+  | AcquireSync1 {_rd :: Bool, _rxEven :: RxEven}
+  | CommaDetect2 {_rd :: Bool, _rxEven :: RxEven}
+  | AcquireSync2 {_rd :: Bool, _rxEven :: RxEven}
+  | CommaDetect3 {_rd :: Bool, _rxEven :: RxEven}
+  | SyncAcquired1 {_rd :: Bool, _rxEven :: RxEven}
+  | SyncAcquired2 {_rd :: Bool, _rxEven :: RxEven}
+  | SyncAcquired2A {_rd :: Bool, _rxEven :: RxEven, _goodCgs :: Index 4}
+  | SyncAcquired3 {_rd :: Bool, _rxEven :: RxEven}
+  | SyncAcquired3A {_rd :: Bool, _rxEven :: RxEven, _goodCgs :: Index 4}
+  | SyncAcquired4 {_rd :: Bool, _rxEven :: RxEven}
+  | SyncAcquired4A {_rd :: Bool, _rxEven :: RxEven, _goodCgs :: Index 4}
+  deriving (Generic, NFDataX, Eq, Show)
+
+-- | Vector containing the 3 code groups that contain a comma (K28.1, K28.5 and
+--   K28.7) and their alternative forms
+commas :: Vec 6 (BitVector 10)
+commas =
+  0b0011111001
+    :> 0b1100000110
+    :> 0b0011111010
+    :> 0b1100000101
+    :> 0b0011111000
+    :> 0b1100000111
+    :> Nil
```

Comment:
You mention 3 codegroups, but I see 6?

Reply:
The alternative forms, so with the opposite running disparity. I have clarified this a bit.